src/twisted/application/app.py

```diff
@@ -501,8 +501,9 @@ def opt_help_reactors(self):
         """
         rcts = sorted(self._getReactorTypes(), key=attrgetter('shortName'))
         for r in rcts:
-            self.messageOutput.write('    %-4s\t%s\n' %
-                                     (r.shortName, r.description))
+            if requireModule(r.moduleName) is not None:
+                self.messageOutput.write('    %-4s\t%s\n' %
+                                         (r.shortName, r.description))
         raise SystemExit(0)
```

Comment:
I second this sentiment:

- the error reporting here is definitely an improvement
- the `!` in the reactor list would be good (maybe a "*" and then at the bottom of "*: reactors not available on this platform, --reactor=whatever for details")
- it need not block this work.